app/coffee/WebsocketController.coffee

```diff
@@ -91,16 +91,29 @@ module.exports = WebsocketController =
 				return callback(error) if error?
 				DocumentUpdaterManager.getDocument project_id, doc_id, fromVersion, (error, lines, version, ranges, ops) ->
 					return callback(error) if error?
+
 					# Encode any binary bits of data so it can go via WebSockets
 					# See http://ecmanaut.blogspot.co.uk/2006/07/encoding-decoding-utf8-in-javascript.html
+					encodeForWebsockets = (text) -> unescape(encodeURIComponent(text))
 					escapedLines = []
 					for line in lines
 						try
-							line = unescape(encodeURIComponent(line))
+							line = encodeForWebsockets(line)
 						catch err
 							logger.err {err, project_id, doc_id, fromVersion, line, client_id: client.id}, "error encoding line uri component"
 							return callback(err)
 						escapedLines.push line
+					if options.encodeRanges
+						try
+							for comment in ranges?.comments or []
+								comment.op.c = encodeForWebsockets(comment.op.c)
+							for change in ranges?.changes or []
+								change.op.i = encodeForWebsockets(change.op.i) if change.op.i
```

Comment:
Actually, one more little thing. We should use `change.op.i?` here. This is an explicit null/undefined check, where as `change.op.i` relies on JS's notion of 'truthy' which can catch us out on say an empty string being 'false'.

Reply:
Right good point. I'll fix